public/app/features/annotations/editor_ctrl.ts

```diff
@@ -56,17 +56,19 @@ export class AnnotationsEditorCtrl {
     this.reset();
 
     this.onColorChange = this.onColorChange.bind(this);
-  }
 
-  async datasourceChanged() {
-    return (this.currentDatasource = await this.datasourceSrv.get(this.currentAnnotation.datasource));
+    $scope.$watch('ctrl.currentAnnotation.datasource', async (value: string) => {
+      const newDatasource = await this.datasourceSrv.get(this.currentAnnotation.datasource);
+      $scope.$apply(() => {
```

Comment:
This was the main fix as the `await` before pushed this assignment outside of $digest cycle (at least I think).